ecosante/api/baignades.py

```diff
@@ -0,0 +1,354 @@
+                ranking_label = ranking_label_from_src(img['src'])
+    if None not in (ranking_year, ranking_label) and is_valid_year(ranking_year):
+        ranking_dict = {
+            "label": ranking_label,
+            "year": ranking_year
+        }
+    return {
+        "season": season_dict,
+        "sample": sample_dict,
+        "ranking": ranking_dict,
+        "interdiction": interdiction_dict,
+        "url": url
+    }
+
+def id_carte_from_insee(insee):
+    return {
+        "971": "gua", # Guadeloupe
+        "972": "mar", # Martinique
+        "973": "guy", # Guyane
+        "974": "reu", # Réunion
+        "976": "may", # Mayotte
+        "975": None,  # COM non supporté
+        "977": None,
+        "978": None,
+        "984": None,
+        "986": None,
+        "987": None,
+        "988": None,
+        "989": None,
+    }.get(insee[0:3], 'fra') # France métropolitaine par défaut
+
+def get_season_year(id_carte, dt):
+    if id_carte == 'fra': # France métropolitaine
+        if dt.month < 6:
+            season_year = str(dt.year - 1) # année précédente en avant-saison
+        else:
+            season_year = str(dt.year) # année en cours dès juin
+    else: # Outre-mer
+        if dt.month < 10:
+            season_year = str(dt.year) # année en cours dès octobre
+        else:
+            season_year = str(dt.year + 1) # année suivante
+    return season_year
+
+def get_site_order(details):
+    interdiction = details['interdiction']
+    if interdiction is not None: # interdiction d'abord
+        order = 0
+    else:
+        order = 10
+    sample_label = details['sample']['label']
+    order += order_from_sample_label(sample_label) # mauvais résultats ensuite
+    return order
+            
+def order_from_sample_label(label):
+    return ["Mauvais résultat", "Résultat moyen", "Bon résultat", "Pas de résultats de prélèvements"].index(label)
+
+def sample_label_from_value(value):
+    return {
+        "Mauvais": "Mauvais résultat",
+        "Moyen": "Résultat moyen",
+        "Bon": "Bon résultat",
+    }.get(value, None)
+
+def ranking_label_from_src(src):
+    if "classe1.gif" in src:
+        ranking_label = "Excellent"
+    elif "classe2.gif" in src:
+        ranking_label = "Bon"
+    elif "classe3.gif" in src:
+        ranking_label = "Suffisant"
+    elif "classe4.gif" in src:
+        ranking_label = "Insuffisant"
+    elif "classe5.gif" in src:
+        ranking_label = "Insuffisamment de prélèvements"
+    elif "classe6.gif" in src or "classe7.gif" in src or "classe11.gif" in src:
+        ranking_label = "Site non classé"
+    else:
+        ranking_label = "Non suivi"
+    return ranking_label
+
+def date_from_str(date_str):
+    day, month, year = date_str.split('/')
+    return date(int(year), int(month), int(day))
+
+def is_valid_date(date_str):
+    is_valid_date = True
+    try:
+        date_from_str(date_str)
+    except ValueError:
+        is_valid_date = False
+    return is_valid_date
```

Comment:
gros pinaillage, parce que ça fonctionne bien comme ça
```suggestion
    try:
        date_from_str(date_str)
    except ValueError:
        return False
    else:
        return True
```